xarray/core/rolling_exp.py

```diff
@@ -79,7 +75,18 @@ def _get_center_of_mass(comass, span, halflife, alpha):
     return float(comass)
 
 
-class RollingExp(Generic[T_DSorDA]):
+# We seem to need to redefine T_Xarray here, rather than importing `core.types`, because
+# a) it needs to be defined (can't be a string) b) it can't be behind an `if
+# TYPE_CHECKING` branch and c) we have import errors if we import it without at the
+# module level like: from .types import T_Xarray
+
+if TYPE_CHECKING:
+    from .dataarray import DataArray
+    from .dataset import Dataset
+T_Xarray = TypeVar("T_Xarray", "DataArray", "Dataset")
```

Comment:
```suggestion
if TYPE_CHECKING:
    from .dataarray import DataArray
    from .dataset import Dataset

from .types import T_Xarray
```
This should work now I think,. Might get linting issues though.